blog-samples/CSharp/FacebookHandover/Primary/Bots/PrimaryBot.cs

```diff
@@ -0,0 +1,213 @@
+                    await turnContext.SendActivityAsync("Passing thread control to the page inbox...");
+                    await FacebookThreadControlHelper.PassThreadControlAsync(_configuration["FacebookPageToken"], PAGE_INBOX_ID, turnContext.Activity.From.Id, text);
+                    break;
+
+                case OPTION_PASS_SECONDARY_BOT:
+
+                    var secondaryReceivers = await FacebookThreadControlHelper.GetSecondaryReceiversAsync(_configuration["FacebookPageToken"]);
+
+                    foreach (var receiver in secondaryReceivers)
+                    {
+                        if (receiver != PAGE_INBOX_ID)
+                        {
+                            await turnContext.SendActivityAsync("Passing thread control to the secondary app...");
+                            await FacebookThreadControlHelper.PassThreadControlAsync(_configuration["FacebookPageToken"], receiver, turnContext.Activity.From.Id, text);
+                            break;
+                        }
+                    }
+
+                    break;
+
+                case OPTION_REQUEST_THREAD_CONTROL:
+                case OPTION_REQUEST_THREAD_CONTROL_NICELY:
+                    // Do nothing because the secondary receiver should react to these instead
+                    break;
+
+                default:
+                    await ShowChoices(turnContext, cancellationToken);
+                    break;
+            }
+        }
+
+        protected override async Task OnConversationUpdateActivityAsync(ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("PrimaryBot - Processing a ConversationUpdate Activity.");
+
+            var facebookPayload = (turnContext.Activity.ChannelData as JObject)?.ToObject<FacebookPayload>();
+
+            if (facebookPayload != null)
+            {
+                if (facebookPayload.PassThreadControl != null)
+                {
+                    await turnContext.SendActivityAsync($"Thread control is now passed to {facebookPayload.PassThreadControl.NewOwnerAppId} with the message: {facebookPayload.PassThreadControl.Metadata}");
+                    await ShowChoices(turnContext, cancellationToken);
+                }
+            }
+
+            await base.OnConversationUpdateActivityAsync(turnContext, cancellationToken);
+        }
+        protected override async Task OnEventActivityAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("PrimaryBot - Processing an Event Activity.");
+
+            // Analyze Facebook payload from EventActivity.Value
+            await ProcessFacebookMessage(turnContext, turnContext.Activity.Value, cancellationToken);
+        }
+
+        private async Task<bool> ProcessFacebookMessage(ITurnContext turnContext, object data, CancellationToken cancellationToken)
```

Comment:
This method is not needed, and the code can be moved into OnEventActivityAsync. Or, at least, the return value can be removed (since it is not used).

Reply:
I'm going to leave this the way it is. The return value actually is used because of a C# quirk.